data/testdata/all_types.golden.json

```diff
@@ -74,8 +74,15 @@
       { "name": "timestamps", "type": "time", "typeInfo": { "frame": "time.Time" }, "config": { "interval": 1000 } },
       { "name": "timestamps", "type": "time", "typeInfo": { "frame": "time.Time" } },
       { "name": "nullable_timestamps", "type": "time", "typeInfo": { "frame": "time.Time", "nullable": true } },
-      { "name": "json", "type":"other", "typeInfo": { "frame": "json.RawMessage" } },
-      { "name": "nullable_json", "type":"other", "typeInfo": { "frame": "json.RawMessage", "nullable": true } }
+      { "name": "json", "type": "other", "typeInfo": { "frame": "json.RawMessage" } },
+      { "name": "nullable_json", "type": "other", "typeInfo": { "frame": "json.RawMessage", "nullable": true } },
+      {
+        "name": "enum",
+        "type": "enum",
+        "typeInfo": { "frame": "enum" },
+        "config": { "mappings": [{ "type": "enum", "options": { "value": ["", "ONE", "TWO", "THREE"] } }] }
```

Comment:
@leeoniya ... does this seem OK?

Reply:
it seems a bit weird that it *can* exist on any field...

Perhaps it would be better to have an `enum` object directly on `FieldConfig` 🤔 that is not standard mapping type

Reply:
my concern with having it be on `config` is that this would get caught up in a deep config diffing and trigger re-init each time. my understanding is that this may not be an exhaustive enum field, but only representative of values that are actually present within the data response, so it can be a partial enum each time the data updates.

https://github.com/grafana/grafana/blob/821614fb431af195d6bd23055272292f42cf68d9/packages/grafana-data/src/dataframe/frameComparisons.ts#L65-L68

Reply:
as far as keeping it in the mappings, would these mappings override any existing mappings or do we combine them with whatever is in the panel config? i think they would blow away any existing mappings?

Reply:
Thinking more generally... this is a good example where we will want config that only applies to a specific type.  We will need something similar for time offset (or similar) eg, the config should say what the absolute start time is and the values are the offset